addon/components/au-badge.js

```diff
@@ -0,0 +1,17 @@
+import Component from '@glimmer/component';
+
+export default class AuBadgeComponent extends Component {
+  get skin() {
+    if (this.args.skin == "info")
```

Comment:
I like to add code that reacts to not having a skin prop, like so `if (this.args.skin == "info" || !args.skin)` so there is a default badge that works, when no skin is provided.

I would consider this grey one to be the default:

![image](https://user-images.githubusercontent.com/12690/79546343-90bb5c80-8092-11ea-8908-e019cc36b112.png)

What do you think? This way the implementation can be 

`<AuBadge @icon="info-circle" />` (can you self-close components?)

Reply:
I have this one as the default (when no skin is added)
<img width="440" alt="Screenshot 2020-05-06 at 15 15 40" src="https://user-images.githubusercontent.com/1572885/81181247-75b27d00-8fac-11ea-8a78-0d5b5d17c69e.png">
:
![Screenshot 2020-05-06 at 15 14 09](https://user-images.githubusercontent.com/1572885/81181145-4b60bf80-8fac-11ea-9ad8-b9afd1d4406f.png)

Which is solved in the css code
It's a different approach, but maybe not standard.
<img width="426" alt="Screenshot 2020-05-06 at 15 14 50" src="https://user-images.githubusercontent.com/1572885/81181192-5ddaf900-8fac-11ea-858a-2b746d108713.png">


What is hte

Reply:
I don't think it's necessary to do a `:not` in your CSS, it's better to not render the class in the first place.

This is all new terroritory for me so, take what I will write here with a grain of salt, I still have to write more components in this Octane style to find the ideal workflow.

I find with this kind of code, you should define every variant in your CSS as a modifier:

```
.c-my-component {}
.c-my-component--default {}
.c-my-component--variant-1 {}
.c-my-component--variant-2 {}
```

This way you don't have to override any styles in your modifier class.

And then when the component is called, and you do not provide an `@skin` prop and thus write something like:

```<MyComponent />```

It should render out the default:

```<div class="c-my-component c-my-component--default"></div>```

To do so in JS one would probably write something like

```
import Component from '@glimmer/component';

export default class MyComponent extends Component {
  get skin() {
    if (this.args.skin == "variant1")
      return "c-my-component--variant1";
    if (this.args.skin == "variant2")
      return "c-my-component--variant2";
    else
      return "c-my-component--default";
  }
}
```